omod/src/main/java/org/openmrs/module/financials/fragment/controller/CumulativeDepartmentalFinanceSummariesFragmentController.java

```diff
@@ -0,0 +1,105 @@
+package org.openmrs.module.financials.fragment.controller;
+
+import org.codehaus.jackson.map.ObjectMapper;
+import org.openmrs.Role;
+import org.openmrs.User;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.financials.GeneralRevenuePerUnit;
+import org.openmrs.module.hospitalcore.HospitalCoreService;
+import org.openmrs.module.hospitalcore.model.OpdTestOrder;
+import org.openmrs.module.hospitalcore.model.PatientServiceBillItem;
+import org.openmrs.ui.framework.SimpleObject;
+import org.openmrs.ui.framework.UiUtils;
+import org.openmrs.ui.framework.fragment.FragmentModel;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import java.math.BigDecimal;
+import java.util.*;
+
+public class CumulativeDepartmentalFinanceSummariesFragmentController {
+	
+	public void controller(FragmentModel model) {
+		
+		model.addAttribute("cashiers", getAllUsersWithRoleCashier());
+	}
+	
+	//cumulativeSummaries
+	public List<SimpleObject> fetchCumulativeSummariesByDateRange(
+	        @RequestParam(value = "cashier", required = false) Integer cashierId,
+	        @RequestParam(value = "fromDate", required = false) String fromDate,
+	        @RequestParam(value = "toDate", required = false) String toDate, UiUtils uiUtils) {
+		List<OpdTestOrder> allOpdOrders = Context.getService(HospitalCoreService.class).getAllOpdOrdersByDateRange(false,
+		    fromDate, toDate);
+		List<PatientServiceBillItem> patientServiceBillItems = Context.getService(HospitalCoreService.class)
+		        .getAllPatientServiceBillItemsByDate(false, fromDate, toDate);
+		GeneralRevenuePerUnit generalRevenuePerUnit = null;
+		List<GeneralRevenuePerUnit> cumulativeSummarries = new ArrayList<GeneralRevenuePerUnit>();
+		
+		for (OpdTestOrder opdTestOrder : allOpdOrders) {
+			for (PatientServiceBillItem patientServiceBillItem : patientServiceBillItems) {
+				if (opdTestOrder.getBillableService().equals(patientServiceBillItem.getService())
+				        && opdTestOrder.getFromDept() != null) {
+					generalRevenuePerUnit = new GeneralRevenuePerUnit();
+					generalRevenuePerUnit.setTransactionDate(opdTestOrder.getScheduleDate());
+					if (opdTestOrder.getConcept().getConceptId() == 1000021) {
```

Comment:
If possible we compare the objects, I mean lets do **opdTestOrder.getConcept().equals(put here a concept object that correspond to that id)**, and try getting a concept by uuid to apply there